java-client/src/main/java/org/opensearch/client/opensearch/cat/IndicesRequest.java

```diff
@@ -44,310 +44,312 @@
+         * A comma-separated list of index names to limit the returned information
+         * <p>
+         * API name: {@code index}
+         * <p>
+         * Adds all elements of <code>list</code> to <code>index</code>.
+         */
+        public final Builder index(List<String> list) {
+            this.index = _listAddAll(this.index, list);
+            return this;
+        }
+
+        /**
+         * A comma-separated list of index names to limit the returned information
+         * <p>
+         * API name: {@code index}
+         * <p>
+         * Adds one or more values to <code>index</code>.
+         */
+        public final Builder index(String value, String... values) {
+            this.index = _listAdd(this.index, value, values);
+            return this;
+        }
+
+        /**
+         * Set to true to return stats only for primary shards
+         * <p>
+         * API name: {@code pri}
+         */
+        public final Builder pri(@Nullable Boolean value) {
+            this.pri = value;
+            return this;
+        }
+
+        @Override
+        protected Builder self() {
+            return this;
+        }
+
+        /**
+         * Builds a {@link IndicesRequest}.
+         *
+         * @throws NullPointerException if some of the required fields are null.
+         */
+        public IndicesRequest build() {
+            _checkSingleUse();
+
+            return new IndicesRequest(this);
+        }
+    }
+
+    // ---------------------------------------------------------------------------------------------
+
+    /**
+     * Endpoint "{@code cat.indices}".
+     */
+    public static final Endpoint<IndicesRequest, IndicesResponse, ErrorResponse> _ENDPOINT = new SimpleEndpoint<>(
+
+            // Request method
+            request -> {
+                return "GET";
+
+            },
+
+            // Request path
+            request -> {
+                final int _index = 1 << 0;
+
+                int propsSet = 0;
+
+                if (ApiTypeHelper.isDefined(request.index()))
+                    propsSet |= _index;
+
+                if (propsSet == 0) {
+                    StringBuilder buf = new StringBuilder();
+                    buf.append("/_cat");
+                    buf.append("/indices");
+                    return buf.toString();
+                }
+                if (propsSet == (_index)) {
+                    StringBuilder buf = new StringBuilder();
+                    buf.append("/_cat");
+                    buf.append("/indices");
+                    buf.append("/");
+                    SimpleEndpoint.pathEncode(request.index.stream().map(v -> v).collect(Collectors.joining(",")), buf);
+                    return buf.toString();
+                }
+                throw SimpleEndpoint.noPathTemplateFound("path");
+
+            },
+
+            // Request parameters
+            request -> {
+                Map<String, String> params = request.queryParameters();
```

Comment:
new HashMap<>(request.queryParameters())
```suggestion
                Map<String, String> params = new HashMap<>(request.queryParameters());
```

Let's not modify the original request

Reply:
ok, thanks